### PR TITLE
Fuse broadcasts with pointwise after splitting the reduction

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: src/fuse_pointwise.cpp [hunk 1]
```diff
@@ -200,8 +200,43 @@ struct pointwise_reshape : rewrite_reshapes_base
     static std::string name() { return "pointwise"; }
 };
 
+struct pointwise_broadcast_pointwise
```
Comment (umangyadav) [inline src/fuse_pointwise.cpp:203]: add unit-tests

File: src/fuse_pointwise.cpp [hunk 2]
```diff
@@ -200,8 +200,43 @@ struct pointwise_reshape : rewrite_reshapes_base
...
+        auto x_ins         = r.instructions["x"];
+
+        auto broadcast = broadcast_ins->get_operator();
+
+        auto x_inputs = x_ins->inputs();
+        std::transform(x_inputs.begin(), x_inputs.end(), x_inputs.begin(), [&](auto input) {
+            return m.insert_instruction(broadcast_ins, broadcast, input);
+        });
+
+        m.replace_instruction(
```
Comment (umangyadav) [inline src/fuse_pointwise.cpp:227]: How does it help to broadcast all the inputs ? 

Is it guranteed that all the inputs are broadcastable ?

Comment (pfultz2) [reply src/fuse_pointwise.cpp:227]: > How does it help to broadcast all the inputs ?

So the output of the operator will have the same dimensions as the original broadcasted instruction which feeds into another pointwise(which all have the same dimesions).

> Is it guranteed that all the inputs are broadcastable ?

Yes, the inputs and outputs are the same dimensions for the pointwise operator so if we can broadcast the output then we can broadcast all the inputs.

File: src/fuse_pointwise.cpp (hunk 1 above)
Comment (pfultz2) [reply src/fuse_pointwise.cpp:203]: Unit test added.

**Review by umangyadav — APPROVED** [review]

**Pull request merged**